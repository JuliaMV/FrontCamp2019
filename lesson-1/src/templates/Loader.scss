.overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: hsla(0, 0%, 13%, 0.5);
}

.overlay_inner {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.overlay_content {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.loader {
  display: inline-block;
  width: 75px;
  height: 75px;
  border: 2px solid hsla(0, 0%, 100%, 0.05);
  border-top-color: hsl(0, 0%, 100%);
  border-radius: 100%;
  animation: spin 1s infinite linear;
}

@keyframes spin {
  100% {
    transform: rotate(360deg);
  }
}
