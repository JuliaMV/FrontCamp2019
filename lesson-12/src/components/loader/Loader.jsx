import React from 'react';

import css from './Loader.module.css';

const Loader = () => (
  <div className={css.overlay}>
    <div className={css.overlay_inner}>
      <div className={css.overlay_content}>
        <span className={css.loader} />
      </div>
    </div>
  </div>
);

export default Loader;
