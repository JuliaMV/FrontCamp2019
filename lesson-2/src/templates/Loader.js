import './Loader.scss';

const Loader = () => `
<div class="overlay">
    <div class="overlay_inner">
        <div class="overlay_content">
          <span class="loader"></span>
        </div>
    </div>
</div>
`;

export default Loader;