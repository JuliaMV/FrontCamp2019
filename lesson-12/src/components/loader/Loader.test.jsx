import React from 'react';
import { create } from 'react-test-renderer';
import Loader from './Loader';

describe('Loader component', () => {
  test('it matches the snapshot', () => {
    const component = create(<Loader />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
