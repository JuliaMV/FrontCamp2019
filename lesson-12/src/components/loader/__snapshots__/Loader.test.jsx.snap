// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component it matches the snapshot 1`] = `
<div>
  <div>
    <div>
      <span />
    </div>
  </div>
</div>
`;
